### ISSUE TITLE
Can't share local folder with diffent user account

### ISSUE THREAD

**Issue by walterfranssen** [docker/for-win#6646]
I update docker desktop today, and when I was running the docker-compose file, my MS-SQL server doesn't work anymore because he can't connect anymore to my c drive.

I'm having an account in azure ad en we use a workaround before with a local user. But the pop-up with user credentials doesn't appear anymore?

#6223 # Expected behavior
That is work like before :)

### Actual behavior
file sharing doens't work any more

### Information
  - Windows Version: Version 1903 (OS Build 18362.778)
  - Docker Desktop Version: Docker version 19.03.8, build afacb8b

### Steps to reproduce the behavior
docker run -e ACCEPT_EULA=Y -e SA_PASSWORD=YourStrong!Passw0rd -v C:\local-dev\data\mssql2:/var/opt/mssql/data mcr.microsoft.com/mssql/server:2017-latest

**logs**
SQL Server 2019 will run as non-root by default.
This container is running as user root.
Your master database file is owned by root.
To learn more visit https://go.microsoft.com/fwlink/?linkid=2099216.
2020-05-12 07:12:40.70 Server      Microsoft SQL Server 2017 (RTM-CU20) (KB4541283) - 14.0.3294.2 (X64)
        Mar 13 2020 14:53:45
        Copyright (C) 2017 Microsoft Corporation
        Developer Edition (64-bit) on Linux (Ubuntu 16.04.6 LTS)
2020-05-12 07:12:40.71 Server      UTC adjustment: 0:00
2020-05-12 07:12:40.71 Server      (c) Microsoft Corporation.
2020-05-12 07:12:40.71 Server      All rights reserved.
2020-05-12 07:12:40.72 Server      Server process ID is 40.
2020-05-12 07:12:40.72 Server      Logging SQL Server messages in file '/var/opt/mssql/log/errorlog'.
2020-05-12 07:12:40.72 Server      Registry startup parameters:
         -d /var/opt/mssql/data/master.mdf
         -l /var/opt/mssql/data/mastlog.ldf
         -e /var/opt/mssql/log/errorlog
2020-05-12 07:12:40.73 Server      Error: 17113, Severity: 16, State: 1.
2020-05-12 07:12:40.73 Server      Error 2(The system cannot find the file specified.) occurred while opening file '/var/opt/mssql/data/master.mdf' to obtain configuration information at startup. An invalid startup option might have caused the error. Verify your startup options, and correct or remove them if necessary.

**Comment by Atrejoe** [comment]
I think I have the same issue: I upgraded just now to Docker Desktop 2.3.0.2, using Docker Engine v19.03.8.
Before upgrading this container ran fine.

Maybe coincidentally also having this issue with SQL Server.

Steps to reproduce:

create file: `.env`
```env
DB_SA_PASSWORD=MySAP@ssw0rd!
```
create file: ` docker-compose.yml`
```docker-compose
version: '3'
services:
  db:
    image: "mcr.microsoft.com/mssql/server"
    environment:
       SA_PASSWORD: "${DB_SA_PASSWORD}"
       ACCEPT_EULA: "Y"
       MSSQL_PID: "Express"
    #Expose port for development purpose only
    ports:
      - 1433:1433
    volumes:
      - ./mylocaldatadrive/:/var/opt/mssql/data
```

start using `docker-compose up -d`
read logs using: `docker-compose logs -f`

```
SQL Server 2019 will run as non-root by default.
db_1  | This container is running as user mssql.
db_1  | Your master database file is owned by root.
db_1  | To learn more visit https://go.microsoft.com/fwlink/?linkid=2099216.
db_1  | 2020-05-12 10:51:10.10 Server      Setup step is FORCE copying system data file 'C:\templatedata\model_replicatedmaster.mdf' to '/var/opt/mssql/data/model_replicatedmaster.mdf'.
2020-05-12 10:51:10.16 Server      ERROR: Setup FAILED copying system data file 'C:\templatedata\model_replicatedmaster.mdf' to '/var/opt/mssql/data/model_replicatedmaster.mdf':  1(Incorrect function.)
db_1  | ERROR: BootstrapSystemDataDirectories() failure (HRESULT 0x80070001)
```

Other volumes I mount from the host are visible, the files can be listed but are shown with a file size of 0 bytes and cannot be copied.

Host info:
Windows 10 (1909) with WSL
Docker running Linux containers on Hyper V

**Comment by Tamrael** [comment]
similar problem here. after upgrade to 2.3.0.2 no folders from windows that were previously mounted appear in the docker container. since the "File sharing" switched from drive-based to folder-based maybe that has something to do with it? I tried to allow whole drive like before, tried to just allow folders...

I gave up after a day of troubleshooting an went back to 2.2.0.5 for now.

**Comment by Atrejoe** [comment]
Certainly feels related to the changed file sharing options, but I guess we'll see.

Downgrading helps (download a non-latest version here: https://docs.docker.com/docker-for-windows/release-notes/ => see the right bar, uninstall old Docker first)

**Comment by stephen-turner** [comment]
More examples in the duplicate ticket #6656.

**Comment by waynesi** [comment]
Since #6656 is closed, move my case here. The issue is very easy to reproduce, just run this -

`docker run --name $Name -e "ACCEPT_EULA=Y" -e "MSSQL_SA_PASSWORD=YourStrong!Passw0rd" -p 1433:1433 -v C:/SQLData/data:/var/opt/mssql/data -v C:/SQLData/log:/var/opt/mssql/log -v C:/SQLData/secrets:/var/opt/mssql/secrets -d mcr.microsoft.com/mssql/server`

My Windows is 1909 (18363.836) **without** WSL (never enabled).

**Comment by djs55** [comment]
Thanks for your reports. @waynesi I tweaked and tried your example from a Linux environment and it fails like this:
```
djs@nuc1:~$ docker run -e "ACCEPT_EULA=Y" -e "MSSQL_SA_PASSWORD=password" -p 1433:1433 -v ~/SQLData/data:/var/opt/mssql/data -v ~/SQLData/log:/var/opt/mssql/log -v ~/SQLData/secrets:/var/opt/mssql/secrets mcr.microsoft.com/mssql/server
SQL Server 2019 will run as non-root by default.
This container is running as user mssql.
To learn more visit https://go.microsoft.com/fwlink/?linkid=2099216.
2020-05-13 10:28:59.18 Server      Setup step is copying system data file 'C:\templatedata\master.mdf' to '/var/opt/mssql/data/master.mdf'.
2020-05-13 10:28:59.27 Server      ERROR: Setup FAILED copying system data file 'C:\templatedata\master.mdf' to '/var/opt/mssql/data/master.mdf':  2(The system cannot find the file specified.)
ERROR: BootstrapSystemDataDirectories() failure (HRESULT 0x80070002)
```
I suspect there might be a bug in the latest image.

Apart from this problem, you might be able to work around any Desktop-specific filesharing issues and increase performance at the same time by using `docker volumes`. Storing database files on the host will always be slow, but if the main requirement is to guarantee persistence between runs then a `docker volume` will achieve that and be much faster because the database files will be stored on the Linux filesystem rather on Windows where they have to be accessed remotely.

For example (with `mariadb`)
```
version: '3.3'
volumes:
  database:
services:
  db:
    image: mariadb:10.3.22
    ports: 
    - "3306:3306"
    environment: 
        MYSQL_ROOT_PASSWORD: password
        MYSQL_DATABASE: mydb
        MYSQL_USER: myuser
        MYSQL_PASSWORD: password     
    volumes:
    - database:/var/lib/mysql  
```

A corresponding example with `mssql` would be:
```
version: '3.3'
volumes:
  secrets:
  data:
services:
  db:
    image: mcr.microsoft.com/mssql/server
    ports:
    - "1433:1433"
    environment:
        ACCEPT_EULA: y
        MSSQL_SA_PASSWORD: YourStrong!Passw0rd
    volumes:
    - secrets:/var/opt/mssql/secrets
    - data:/var/opt/mssql/data
    - ./log:/var/opt/mssql/log
```
(Note this fails for me with the same error that I see on Linux native)

**Comment by AlphaRage** [comment]
Just gonna mention that I have this issue, too, as noted in my (now closed) duplicate. #6660 
Does not matter if you use docker-compose or docker run. If you mount a volume on (in my case) an MSSQL-Server Container it will crash.

**Comment by Atrejoe** [comment]
@djs55 thanks for your suggestion. It may be best practice to use proper volumes, and a great tip, but there are more possible reasons for sharing host volumes. (In my case restoring backups).

Can you share what version of the Docker engine you are using?

Looking at the releases of the images of SQL server, this is unlikely to be due to a new build of the SQL server image (example also mix 2019 and 2017 images).
https://hub.docker.com/_/microsoft-mssql-server?tab=description

**Comment by djs55** [comment]
@Atrejoe My Linux docker version is 19.03.6 which is a little older than the version bundled in Desktop: 19.03.8

**Comment by thaJeztah** [comment]
When you're bind-mounting a directory using the shorthand `-v <host-path>:<container-path>` syntax, and the directory does not exist on the host where the daemon runs, the docker daemon creates the `<host-path>` directory before starting the container. In that case, the daemon creates the directory with default ownership (and permissions), which (given that the daemon itself runs as `root`) will be `root:root`.

In the Linux situation, that means the container may fail, because the data-directory is owned by `root` (and doesn't match the container user). To prevent that situation on Linux (if you really can't use a managed volume), is to make sure the files on the host match the uid/gid of the user inside the container (Linux uses the numeric IDs of users/groups: the name doesn't matter, so the user doesn't have to _exist_ on the host as long as the numeric IDs match).

On Docker Desktop, the situation is slightly different: Docker Desktop "ignores" ownership when a process checks the owner of a file / directory that's bind-mounted. Instead it always shows the user "checking" the ownership as being the owner. This feature is to assist developers to allow them to "access" files on their host, even if they're created by `root` by the container (which would otherwise prevent you from accessing those files).

In these examples you can see that "whoever is asking" is listed as the `owner` of `bar`;

```bash
$ docker run -it --rm -v $(pwd):/foo/bar -u 1234:1234 alpine sh -c 'ls -la /foo'
total 8
drwxr-xr-x    3 root     root          4096 May 13 11:56 .
drwxr-xr-x    1 root     root          4096 May 13 11:56 ..
drwxr-xr-x    3 1234     1234            96 May 13 11:40 bar

$ docker run -it --rm -v $(pwd):/foo/bar -u 4321:4321 alpine sh -c 'ls -la /foo'
total 8
drwxr-xr-x    3 root     root          4096 May 13 11:56 .
drwxr-xr-x    1 root     root          4096 May 13 11:56 ..
drwxr-xr-x    3 4321     4321            96 May 13 11:40 bar
```


I'm not familiar with how the MS SQL container image is set up, but if the container _itself_ is started as `root` to do some bootstrapping and later drops to a non-privileged user, it's possible that the bootstrap script checks permissions (as `root`, and hence seeing the file being owned by `root`).

**Comment by waynesi** [comment]
Hi @djs55 thanks for your comments. I can't comment much because my environment is Windows OS whilst yours is Linux and I'm not sure the docker code concerned is same, although I saw same error. As @Atrejoe said, there are many reasons we have to use host volumes.

**Comment by AlphaRage** [comment]
@thaJeztah Everything you're saying is correct as far as I know.
The Container for the 2017 MSSQL-Server runs as user root by default while the 2019 one runs as non-root user mssql for security reasons. The folder access rights apply. But the issue described here appears to be unrelated to access rights.
I tested Docker Desktop 2.3.0.2 with both the 2017 and 2019 version. They both crash (although I am not sure right now if the error message was the same).

**Comment by waynesi** [comment]
@AlphaRage I also tried to run SQL 2019 image as `root` user following MS doc using `-u 0:0`. From the container log, I did see it reported running as root user, however it still failed.

**Comment by thaJeztah** [comment]
> while the 2019 one runs as non-root user mssql for security reasons

@AlphaRage @waynesi can you `docker image inspect` the image? Wondering if the container itself defaults to running as non-root, or if it starts a s root but drops privileges after that (just for completeness; it may not be the cause as mentioned above)

**Comment by waynesi** [comment]
@thaJeztah  here you are ...

[
    {
        "Id": "sha256:d60e9ac97708e7568c2ce3bd2a282ee66022540f1cb815cb282f27512a3c44fd",
        "RepoTags": [
            "mcr.microsoft.com/mssql/server:latest"
        ],
        "RepoDigests": [
            "mcr.microsoft.com/mssql/server@sha256:360f6e6da94fa0c5ec9cbe6e391f411b8d6e26826fe57a39a70a2e9f745afd82"
        ],
        "Parent": "",
        "Comment": "",
        "Created": "2020-03-26T23:22:48.267165811Z",
        "Container": "c46f0129eee5a97e7a7b37a530cb0008dac0ea9ae9b2534ad3b553a23a21516d",
        "ContainerConfig": {
            "Hostname": "d51a9a74169c",
            "Domainname": "",
            "User": "mssql",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "1433/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "#(nop) ",
                "CMD [\"/opt/mssql/bin/sqlservr\"]"
            ],
            "ArgsEscaped": true,
            "Image": "sha256:4fc89bbedc0a52733a57b4e4293d32330ea418327449f14862a3c5ea41b04a03",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/opt/mssql/bin/permissions_check.sh"
            ],
            "OnBuild": [],
            "Labels": {
                "com.microsoft.product": "Microsoft SQL Server",
                "com.microsoft.version": "15.0.4033.1",
                "vendor": "Microsoft"
            }
        },
        "DockerVersion": "1.13.1",
        "Author": "dpgswdist@microsoft.com",
        "Config": {
            "Hostname": "d51a9a74169c",
            "Domainname": "",
            "User": "mssql",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "1433/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "/opt/mssql/bin/sqlservr"
            ],
            "ArgsEscaped": true,
            "Image": "sha256:4fc89bbedc0a52733a57b4e4293d32330ea418327449f14862a3c5ea41b04a03",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/opt/mssql/bin/permissions_check.sh"
            ],
            "OnBuild": [],
            "Labels": {
                "com.microsoft.product": "Microsoft SQL Server",
                "com.microsoft.version": "15.0.4033.1",
                "vendor": "Microsoft"
            }
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 1501323471,
        "VirtualSize": 1501323471,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/9d16ca488e7861e70047c9f7f99e3fe886546da309384ac61b2bd3bff74b0956/diff:/var/lib/docker/overlay2/b7624126d4e06e249b2c634e4f49e4d5ad38f70a0432bfbc1928a706bea7eb41/diff:/var/lib/docker/overlay2/fbfb3f921ef7df388d1accc9ccb4e151bfcb42b466a24f1497f8a24274ea4656/diff:/var/lib/docker/overlay2/ae1d2beacaf559187ca5024b39744a6c1a60040273c7984a2cc4445531f49707/diff:/var/lib/docker/overlay2/dd59967e1315e55f4ed76ef6c7e08e36e4f47c082b42170b351e4c5aeb143559/diff:/var/lib/docker/overlay2/c603e9b816afc082b594f5e32128b2e30f125197645bf336c2010da2862d1694/diff:/var/lib/docker/overlay2/1ff5c10737bdf5c35dc0726a0f13bcfdf4221315f7031bfb54a053093587be23/diff",
                "MergedDir": "/var/lib/docker/overlay2/5435e56570f2968330f07f00ebed6b1ecfa2c8da19de2b8a0c1c613f7104ebcf/merged",
                "UpperDir": "/var/lib/docker/overlay2/5435e56570f2968330f07f00ebed6b1ecfa2c8da19de2b8a0c1c613f7104ebcf/diff",
                "WorkDir": "/var/lib/docker/overlay2/5435e56570f2968330f07f00ebed6b1ecfa2c8da19de2b8a0c1c613f7104ebcf/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:ba9de9d8475e7f5e40086358a1353b3cc080994fc6d31e4272dd3acb69b0151e",
                "sha256:fbd2732ad777cb5db2515fa62b6122b797be233d01db02e0a19e5d894688cad6",
                "sha256:dda1518598187bf87704acc22aa0ec2a67d9e7835c24346dfca118ab42c5cd0b",
                "sha256:75e70aa52609fdbd63b58d46d6f7c20470062e1c9bb75616f7703a358a61e5ca",
                "sha256:1d879a61f0e0c57fddf41eba54272fcc68d9ce2072533ca32091f8f55959f0c8",
                "sha256:800a9a3ba364c39c237d351fc024dc055bf9a3e354716a01d2d8d55857f06cbf",
                "sha256:b2ac6ae4facb4883163bfd804d86da3bbaf21c32c0ed0783c170f1b8106288f9",
                "sha256:ff8bea33f654d8c2d71ca1010c9e66b12c482ffbdb9a7555c3f91cdabe419078"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]

**Comment by thaJeztah** [comment]
Thanks! So from that output;

- Microsoft uses a very, _very_ old version of Docker to build the image (`"DockerVersion": "1.13.1",` 😱)
- The image is configured to run as `mssql` by default (`"User": "mssql",`) so not the `root` -> drop to `mssql` situation

So no direct idea (other than the recommendation to use volumes instead of bind-mounts) 😅

**Comment by waynesi** [comment]
@thaJeztah , bit off-topic, does Docker version matter in terms of building images? I have a few in-house images built a couple of years ago. They're serving us well atm and your comments got me wonder if I'd better rebuild them.

**Comment by thaJeztah** [comment]
Overall it should be fine (images should be compatible in all cases, unless _really_ old (before Docker 1.10 I think, which used older specs, and may have some issues). Just be careful with those machines to only run them in a secure environment, because the engine itself may/will have unpatched vulnerabilities (at least that I know of for Linux) so don't run untrusted load, and don't have them accessible in any way outside of your secure environment.

**Comment by Tamrael** [comment]
@thaJeztah I'm using an image based on php:7.3 build 1 month ago and have no files in the vagrant folder after vagrant up on the new version of docker but getting all files just fine in the previous version.. I tried to mount as extra volume, I tried to add the direct folder, drive and parent folder. nothing help to get the files inside the container on 2.3.0.2. any ideas?

**Comment by thaJeztah** [comment]
Not sure I understand; why are you using vagrant? Or is vagrant spinning up a VM in which you have Windows (in which runs a VM to run Linux)? 🙃🤓

**Comment by archonandrewhunt** [comment]
Same problem when using the latest SQL Server 2019 linux image "mcr.microsoft.com/mssql/server:2019-latest". I was regularly sharing a local folder through volume mounting and today trying on 2.3.0.2 it no longer works.

**Comment by saulyt** [comment]
Having the same issue with server:2019-latest as well.

**Comment by bzylstra** [comment]
We are seeing a similar issue where we use a volume to get a backup available to the SQL Server container.
  mssql:
    image: microsoft/mssql-server-linux:2017-latest
    ports:
    - "1433:1433"
    environment:
      SA_PASSWORD: "*********"
      ACCEPT_EULA: "Y"
      MSSQL_MEMORY_LIMIT_MB: "500"
      MSSQL_AGENT_ENABLED: "true"
    volumes:
    - ./TestDatabase/pa_backup:/db_backup

The container starts up fine but a subsequent attempt to restore from a file on the volume fails with file not found. 
"RESTORE DATABASE MyDb FROM DISK = '/db_backup/MyDb.bak' WITH MOVE 'MyDb' TO '/var/opt/mssql/data/MyDb.mdf', MOVE 'MyDb_log' to '/var/opt/mssql/data/MyDb_log.ldf';"

I also tried to use SSMS to perform the restore and it could not find any files in the volume mounted /db_backup.

I shelled into the SqlServer container with docker-compose exec mssql /bin/bash and was able to view the contents of /db_backup. I was also able to copy MyDb.bak to another folder that was not on the mounted volume and then could perform the restore.

It has the feel of a permissions problem but specifying user: root did not help.

I have reverted to Docker for Windows 2.2.0.4 and things are working again.

**Comment by Tamrael** [comment]
> Not sure I understand; why are you using vagrant? Or is vagrant spinning up a VM in which you have Windows (in which runs a VM to run Linux)?

Because vagrant spins up the docker containers needed for our app and we switched from virtual box with vagrant to docker with vagrant a while back. Vagrant just issues docker commands, no VM. I'm in windows with docker desktop on hyper-v. vagrant shouldn't be the a factor because the volume gets bound and shows as overlay but the content doesn't make it to the container at all. Vagrant uses the -v syntax which from all I've gathered in this thread should work just fine for windows because there is no permissions problem. Containers start just fine but the content is missing inside

**Comment by Atrejoe** [comment]
Maybe interesting to notice, like shown in the image in https://github.com/docker/for-win/issues/6646#issuecomment-627272165, SQL does create *some* files on the shared volume.
It can also list some files, but not access the content nor read the length.

**Comment by waynesi** [comment]
@Atrejoe yes your observation is very correct. When binding an empty folder in my case, a master.mdf of size 0 was created.

**Comment by mat007** [comment]
I can confirm that following the instructions at https://github.com/docker/for-win/issues/6646#issuecomment-627272165 I was able to run it with 2.2.0.5 but not with 2.3.0.2.

**Comment by Atrejoe** [comment]
@mat007 does the fix mentioned in https://github.com/docker/for-win/issues/6680#issuecomment-629256398 also possibly address this issue?

**Comment by mat007** [comment]
@Atrejoe it doesn’t as this is a separate issue, but we have a fix for this one here now in a new private build if you’d like to try it: https://download-stage.docker.com/win/stable/45351/Docker%20Desktop%20Installer.exe

**Comment by miguelelvir** [comment]
Perhaps related to this also? https://github.com/microsoft/mssql-docker/issues/606

**Comment by vkappie** [comment]
@mat007 Thx, this staging build works OK for me. But is there a fix for MacOS version ? because I have problems with volumes access there too ?

**Comment by dtkujawski** [comment]
@mat007 Thanks for the private build - this release fixed my issue with SQL Server and volumes.

**Comment by mat007** [comment]
@vkappie what kind of problems exactly on Mac?
Here is the same build as above but for the Mac version, but I have no idea if it has a chance of fixing your issues without knowing a bit more about them. It may be worth a try though: https://download-stage.docker.com/mac/stable/45351/Docker.dmg

**Comment by Atrejoe** [comment]
@mat007 yes, your private build appears to have fixed the issue. SQL server works in this build again.
I haven't done extensive testing with other images.

**Comment by Tamrael** [comment]
@mat007 can confirm that the private build you shared works as intended on windows and mounts volumes inside docker containers started by vagrant on docker for windows. Thank you for the build and the fix

**Comment by vkappie** [comment]
@mat007 I think it's the same as this one 
"Error 87(The parameter is incorrect.) occurred while opening file '/var/opt/mssql/data/master.mdf' to obtain configuration information at startup. An invalid startup option might have caused the error. Verify your startup options, and correct or remove them if necessary."
Data folder is mapped to internal volume (inside docker VM)
But internal build doesn't work for MacOS. I think I'll create new ticket in MacOS repo. 
Thanks

**Comment by TobiasCortese** [comment]
I am unable to mount a volume with the latest stable build (2.3.0.2) which I believe is related to this issue. The problem is fixed in the latest Edge Build 2.3.1.0. 

In general, it's unsettling that the issue was first unidentified within a day of the release of the latest stable build (May 11) but Docker has yet to release a Stable build update and the issue doesn't appear to be listed in the known issues for the current Stable build. 

So, to ease my mind and instill confidence, could someone from Docker please explain the delay and the lack of communication?  

Here is my scenario, which does not mount with the current Stable build but does mount with the current Edge build...

Dockerfile
FROM continuumio/anaconda3
EXPOSE 48080
COPY requirements.txt /
RUN pip install -r /requirements.txt
CMD ["jupyter", "lab", "--ip='0.0.0.0'", "--port=48080", "--no-browser", "--allow-root"]

$ docker build -t jupyterlab:latest .
$ docker run --name jupyterlab --rm -p 48080:48080 -v "/$(pwd)"://home/ jupyterlab

**Comment by stephen-turner** [comment]
Replied on #6680.

**Comment by eduarantes** [comment]
Docker release version 2.3.0.3 (45519) works for me. Updated today.
![image](https://user-images.githubusercontent.com/4731238/83055180-e12ac000-a029-11ea-9fc3-72dac74ad094.png)

**Comment by mat007** [comment]
Indeed, the fix has now been released in Docker Desktop Stable 2.3.0.3, I’ll go ahead and close this issue.

**Comment by Nuschler** [comment]
Anyone else still seeing these errors after updating to Docker Desktop 2.3.0.3 (45519) or is it just me?
